### PR TITLE
add quantized_resize and dequantize for some cuda backends

### PR DESCRIPTION
**Pull request by hyuen**
Test Plan: added tests with the diff, verifying the windows failures

Differential Revision: D29494134

### CODE REVIEW

**Comment by hyuen** [comment]
The failure in windows is environmental, it failed to install vscode

File: aten/src/ATen/native/native_functions.yaml [hunk 1]
```diff
@@ -7243,6 +7245,7 @@
     CPU: cpu_equal
     CUDA: cuda_equal
     QuantizedCPU: equal_quantized_cpu
+    QuantizedCUDA: cuda_equal_quantized
```
Comment (supriyar) [inline aten/src/ATen/native/native_functions.yaml:7248]: should we name this `equal_quantized_cuda` to be consistent with above?

File: aten/src/ATen/native/quantized/cuda/base_functions.cu [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+#include <ATen/native/DispatchStub.h>
+#include <ATen/native/ReduceOps.h>
+#include <ATen/native/SharedReduceOps.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/cuda/Reduce.cuh>
+
+namespace at {
+namespace native {
+
+bool cuda_equal_quantized(const Tensor& self, const Tensor& src) {
```
Comment (supriyar) [inline aten/src/ATen/native/quantized/cuda/base_functions.cu:11]: Are we checking for equality of quantized tensors here? I think we should also check the qscheme and scale/zero_point in that case.
cc @jerryzh168 looked at something similar for all_close for quantized tensors

File: test/quantization/core/test_quantized_tensor.py [hunk 3]
```diff
@@ -864,6 +864,66 @@ def test_qtensor_resize(self):
             q3 = q1.resize(*sizes2)
             q4 = q1.contiguous().resize(*sizes2)
 
+    @unittest.skipIf(not torch.cuda.is_available() or TEST_WITH_ROCM, 'CUDA is not available')
+    def test_cuda_cpu_qimplementation_consistency(self):
```
Comment (supriyar) [inline test/quantization/core/test_quantized_tensor.py:868]: Is this a general test for testing quantize_per_tensor/dequantize on cpu/cuda? It doesn't seem related to the functions added above, unless I'm missing something

File: test/quantization/core/test_quantized_tensor.py [hunk 4]
```diff
@@ -864,6 +864,66 @@ def test_qtensor_resize(self):
...
+        q1_int = torch.randint(0, 100, sizes1, dtype=dtype, device='cpu')
+        q1 = torch._make_per_tensor_quantized_tensor(a_int.cuda(), scale=scale, zero_point=zero_point)
+        err_str = "requested resize to*"
+        with self.assertRaisesRegex(RuntimeError, err_str):
+            q2 = q1.resize(*sizes1[:-1])
+        # resize on both contiguous and non-contiguous tensor should be fine
+        q3 = q1.resize(*sizes2)
+        q4 = q1.contiguous().resize(*sizes2)
+
+
```
Comment (supriyar) [inline test/quantization/core/test_quantized_tensor.py:926]: Can we also add test for `equal` and `empty_strided`?

File: aten/src/ATen/native/native_functions.yaml [hunk 5]
```diff
@@ -1796,7 +1796,7 @@
   dispatch:
     CPU, Meta: resize_
     CUDA: resize_cuda_
-    QuantizedCPU: quantized_resize_cpu_
+    QuantizedCPU, QuantizedCUDA: quantized_resize_cpu_
```
Comment (jerryzh168) [inline aten/src/ATen/native/native_functions.yaml:1799]: following naming convention for quantized functions, this should be resize_quantized_ I think

File: aten/src/ATen/native/native_functions.yaml [hunk 6]
```diff
@@ -1816,6 +1816,7 @@
   dispatch:
     CPU: empty_strided_cpu
     CUDA: empty_strided_cuda
+    QuantizedCUDA: empty_strided_cuda_quantized
```
Comment (jerryzh168) [inline aten/src/ATen/native/native_functions.yaml:1819]: nit: empty_strided_quantized_cuda

the naming convention is function_name_{backend_name}
